src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (objects.length == size) {
+            increaseObjectsLength();
+        }
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateIndex(index);
+        Object[] increasedObjects = new Object[objects.length + 1];
+        if (index > 0) {
+            System.arraycopy(objects, 0, increasedObjects, 0, index);
```

Comment:
it should be enough to use instead of 31-34 line 
```
if (objects.length == size) {
            increaseObjectsLength();
        }
```

Reply:
done